api/v1/db/json/notes-dao.js

```diff
@@ -0,0 +1,93 @@
+const appRoot = require('app-root-path');
+const _ = require('lodash');
+const fs = require('fs');
+const config = require('config');
+
+const { serializeNotes } = require('../../serializers/notes-serializer');
+
+/**
+ * @summary Return a list of notes filtered/sorted by query parameters
+ * @function
+ * @param {Object} query Query parameters
+ * @returns {Promise} Promise object represents a list of notes
+ */
+const getNotes = query => new Promise((resolve, reject) => {
+  try {
+    const { dbDirectoryPath } = config.api;
+    if (!fs.existsSync(`${appRoot}${dbDirectoryPath}`)) {
+      reject(new Error(`DB directory path: '${dbDirectoryPath}' is invalid`));
+    }
+
+    const {
+      studentID, q, sortKey, sources, contextTypes,
+    } = query;
+    const dirPath = `${dbDirectoryPath}/${studentID}`;
+    let files;
+    let rawNotes = [];
+    try {
+      files = fs.readdirSync(`${appRoot}${dirPath}`);
+    } catch (ignore) {
+      // rawNotes should remain an empty array if directory does not exist
+    }
+
+    _.forEach(files, (file) => {
+      rawNotes.push(appRoot.require(`${dirPath}/${file}`));
+    });
+
+    if (contextTypes) {
+      const contexts = contextTypes.toString().split(',');
+      rawNotes = _.filter(rawNotes, it => _.includes(contexts, it.context.contextType));
+    }
+
+    if (q) {
+      rawNotes = _.filter(rawNotes, it => _.includes(it.note, q));
+    }
```

Comment:
I guess we can use ternary operators here:

```suggestion
     rawNotes = q ? _.filter(rawNotes, it => _.includes(it.note, q)) : rawNotes;
```

Reply:
Nice, looks much cleaner